lcm/lcm.c

```diff
@@ -233,6 +233,13 @@ lcm_handle_timeout (lcm_t *lcm, int timeout_milis)
   timeout.tv_sec = timeout_milis / 1000;
   timeout.tv_usec = (timeout_milis % 1000) * 1000;
 
+// Select in Win32 does not return error on negative timeouts.
+#ifdef WIN32
```

Comment:
this seems like a reasonable thing to do regardless of platform. suggest removing the #ifdef and run the check on all platforms.